apps/system/dat/dat_main.c

```diff
@@ -0,0 +1,326 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * examples/dat/dat_main.c
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/types.h>
+#include <stdio.h>
+
+#include <sys/ioctl.h>
+#include <tinyara/clock.h>
+#include <tinyara/os_api_test_drv.h>
+#include <apps/system/dat_init.h>
+
+#include <apps/shell/tash.h>
+
+/****************************************************************************
+ * Name:  show_usage
+ *
+ * Description:
+ *   This function prints the usage of dat tash command
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+
+static void show_usage(FAR const char *progname)
```

Comment:
Is progname required here as its not used in the function below and the description also says no inputs.

Reply:
modified